server/sonar-server/src/test/java/org/sonar/server/db/migrations/v51/FeedSemaphoresLongDatesTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.db.migrations.v51;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.sonar.api.utils.System2;
+import org.sonar.core.persistence.DbTester;
+import org.sonar.server.db.migrations.DatabaseMigration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.sonar.api.utils.DateUtils.parseDate;
+
+public class FeedSemaphoresLongDatesTest {
+  @ClassRule
+  public static DbTester db = new DbTester().schema(FeedSemaphoresLongDatesTest.class, "schema.sql");
+
+  @Before
+  public void before() throws Exception {
+    db.prepareDbUnit(getClass(), "before.xml");
+  }
+
+  @Test
+  public void execute() throws Exception {
+    DatabaseMigration migration = newMigration(System2.INSTANCE);
+
+    migration.execute();
+
+    int count = db
+      .countSql("select count(*) from semaphores where " +
+        "created_at_ms is not null " +
+        "and updated_at_ms is not null " +
+        "and locked_at_ms is not null");
+    assertThat(count).isEqualTo(2);
+  }
+
+  @Test
+  public void take_now_if_date_in_the_future() throws Exception {
+    DatabaseMigration migration = newMigration(mock(System2.class));
+
+    migration.execute();
+
+    int count = db
+      .countSql("select count(*) from semaphores where " +
+        "created_at_ms = 0");
```

Comment:
Using 0 for now() is misleading. It could be a bug for example if System2#now() is not called. I would mock System2#now() and return an hardcoded value.